uuid.go

```diff
@@ -66,6 +66,10 @@ func Parse(s string) (UUID, error) {
 	switch len(s) {
 	// xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
 	case 36:
+		// Ensure proper dash placement
```

Comment:
I do not see any benefit in these changes.  Checking for "-" is already done on lines 94-89.